### ISSUE TITLE
Stencil exception

### ISSUE THREAD

**Issue by seibert** [numba/numba#2580]
I was testing the still-in-progress stencil PR and ran into this:

``` python
import numba
import numpy as np

N = 10

@numba.jit(nopython=True, parallel=True)
def blur(a):    
    def stencil_kernel(a):
        acc = 0.0
        for i in range(-N, N+1):
            for j in range(-N, N+1):
                acc = a[i,j]

        return acc/((N+1)*(N+1))

    return numba.stencil(stencil_kernel, neighborhood=((-N,N),(-N,N)))(a)

test = np.arange(10000).reshape((100,100))
blur(test)
```
output:
```
---------------------------------------------------------------------------
TypeError                                 Traceback (most recent call last)
<ipython-input-65-07233eabce12> in <module>()
     14 
     15 test = np.arange(10000).reshape((100,100))
---> 16 blur(test)

/Users/sseibert/anaconda/lib/python2.7/site-packages/numba-0.36.0.dev0+264.g616bb493a-py2.7-macosx-10.6-x86_64.egg/numba/dispatcher.pyc in _compile_for_args(self, *args, **kws)
    305                 argtypes.append(self.typeof_pyval(a))
    306         try:
--> 307             return self.compile(tuple(argtypes))
    308         except errors.TypingError as e:
    309             # Intercept typing error that may be due to an argument

/Users/sseibert/anaconda/lib/python2.7/site-packages/numba-0.36.0.dev0+264.g616bb493a-py2.7-macosx-10.6-x86_64.egg/numba/dispatcher.pyc in compile(self, sig)
    577 
    578                 self._cache_misses[sig] += 1
--> 579                 cres = self._compiler.compile(args, return_type)
    580                 self.add_overload(cres)
    581                 self._cache.save_overload(sig, cres)

/Users/sseibert/anaconda/lib/python2.7/site-packages/numba-0.36.0.dev0+264.g616bb493a-py2.7-macosx-10.6-x86_64.egg/numba/dispatcher.pyc in compile(self, args, return_type)
     78                                       impl,
     79                                       args=args, return_type=return_type,
---> 80                                       flags=flags, locals=self.locals)
     81         # Check typing error if object mode is used
     82         if cres.typing_error is not None and not flags.enable_pyobject:

/Users/sseibert/anaconda/lib/python2.7/site-packages/numba-0.36.0.dev0+264.g616bb493a-py2.7-macosx-10.6-x86_64.egg/numba/compiler.pyc in compile_extra(typingctx, targetctx, func, args, return_type, flags, locals, library)
    761     pipeline = Pipeline(typingctx, targetctx, library,
    762                         args, return_type, flags, locals)
--> 763     return pipeline.compile_extra(func)
    764 
    765 

/Users/sseibert/anaconda/lib/python2.7/site-packages/numba-0.36.0.dev0+264.g616bb493a-py2.7-macosx-10.6-x86_64.egg/numba/compiler.pyc in compile_extra(self, func)
    358         self.lifted = ()
    359         self.lifted_from = None
--> 360         return self._compile_bytecode()
    361 
    362     def compile_ir(self, func_ir, lifted=(), lifted_from=None):

/Users/sseibert/anaconda/lib/python2.7/site-packages/numba-0.36.0.dev0+264.g616bb493a-py2.7-macosx-10.6-x86_64.egg/numba/compiler.pyc in _compile_bytecode(self)
    720         """
    721         assert self.func_ir is None
--> 722         return self._compile_core()
    723 
    724     def _compile_ir(self):

/Users/sseibert/anaconda/lib/python2.7/site-packages/numba-0.36.0.dev0+264.g616bb493a-py2.7-macosx-10.6-x86_64.egg/numba/compiler.pyc in _compile_core(self)
    707 
    708         pm.finalize()
--> 709         res = pm.run(self.status)
    710         if res is not None:
    711             # Early pipeline completion

/Users/sseibert/anaconda/lib/python2.7/site-packages/numba-0.36.0.dev0+264.g616bb493a-py2.7-macosx-10.6-x86_64.egg/numba/compiler.pyc in run(self, status)
    244                     # No more fallback pipelines?
    245                     if is_final_pipeline:
--> 246                         raise patched_exception
    247                     # Go to next fallback pipeline
    248                     else:

TypeError: Caused By:
Traceback (most recent call last):
  File "/Users/sseibert/anaconda/lib/python2.7/site-packages/numba-0.36.0.dev0+264.g616bb493a-py2.7-macosx-10.6-x86_64.egg/numba/compiler.py", line 238, in run
    stage()
  File "/Users/sseibert/anaconda/lib/python2.7/site-packages/numba-0.36.0.dev0+264.g616bb493a-py2.7-macosx-10.6-x86_64.egg/numba/compiler.py", line 494, in stage_parfor_pass
    parfor_pass.run()
  File "/Users/sseibert/anaconda/lib/python2.7/site-packages/numba-0.36.0.dev0+264.g616bb493a-py2.7-macosx-10.6-x86_64.egg/numba/parfor.py", line 252, in run
    stencil_pass.run()
  File "/Users/sseibert/anaconda/lib/python2.7/site-packages/numba-0.36.0.dev0+264.g616bb493a-py2.7-macosx-10.6-x86_64.egg/numba/stencilparfor.py", line 81, in run
    arg_to_arr_dict)
  File "/Users/sseibert/anaconda/lib/python2.7/site-packages/numba-0.36.0.dev0+264.g616bb493a-py2.7-macosx-10.6-x86_64.egg/numba/stencilparfor.py", line 134, in _mk_stencil_parfor
    arg_to_arr_dict)
  File "/Users/sseibert/anaconda/lib/python2.7/site-packages/numba-0.36.0.dev0+264.g616bb493a-py2.7-macosx-10.6-x86_64.egg/numba/stencilparfor.py", line 359, in _replace_stencil_accesses
    list(index_list), new_body, scope, loc)
TypeError: 'Var' object is not iterable

Failed at nopython (convert to parfors)
'Var' object is not iterable
```

**Comment by seibert** [comment]
@DrTodd13: Ideas what I'm doing wrong here?

**Comment by DrTodd13** [comment]
We overlooked something in the stencilparfor code.  I have fixed it.  Try it again.

It also seems to work if I remove the @numba.njit altogether on blur but it fails if it is @numba.njit without parallel=True with the error "unsupported keyword arguments when calling Function(<function stencil at 0x7f3682001e18>)".  The stencil function uses **options so I guess that is why regular Numba njit doesn't know it takes a neighborhood parameter?
